be/src/olap/rowset/segment_v2/segment.cpp

```diff
@@ -472,31 +510,26 @@ Status Segment::new_column_iterator_with_path(const TabletColumn& tablet_column,
     // then we could optimize to generate a default iterator
     // This file doest not contain this column, so only read from sparse column
     // to avoid read amplification
```

Comment:
add comment for each branch: leaf node, mid node with data, mid node without data

Reply:
done